### PR TITLE
INFILTRATION: Added new faction called infiltrators that provide infiltration specific augs.

### LINKED ISSUE #3296: CotMG has NF

### PR DESCRIPTION
**Pull request by phyzical**
Adding infiltrators faction

* adds mini game specific augmentations
* adds rep reward to this new faction upon completing infilitration
* reward is relative to the starting security level hardest rewards 10 as base easiest rewards 1.7 as base
* two augmentations to increase this base reward by 5 and multiple it by 2
* fixed bugged with double tick reward on wire cutting game
* bunch of refactors handled by linked prs
* added augmentation for increased money reward for selling information
* added augmentation for increased rep reward when trading
* added augmentation to increase time for for rounds
* added augmentation to decrease health lost
* added new ns function calculateDifficulty
* added new ns function calculateRewards
* added new ns function getLocations
* added new ns function getInfiltrations
* hides multiplier for infil cause its relative to cost

closes #3296

TODO
* add migration for new faction
* update lore for infiltrators as its meh

### CODE REVIEW

**Comment by hydroflame** [comment]
PR made before mega fmt

run `npm run format` and push to (probably) fix merge conflicts.

File: src/ScriptEditor/NetscriptDefinitions.d.ts [hunk 1]
```diff
@@ -4198,6 +4198,59 @@ interface Stanek {
...
+   */
+  calculateRewards(locationName: string): InfiltrationReward;
+  /**
+   * Get all locations that can be infiltrated.
+   * @remarks
+   * RAM cost: 5 GB
+   *
+   * @returns all locations that can be infiltrated.
+   */
+  getLocations(): any[];
```
[the remark this answers is not in the document]
Comment (phyzical) [reply src/ScriptEditor/NetscriptDefinitions.d.ts:4265]: not sure why i did many maybe Location was being mean i can try again

[the remark this answers is not in the document]
Comment (phyzical) [reply src/ScriptEditor/NetscriptDefinitions.d.ts:4301]: right so yeah the reason its any is due to Location not being a type that can be applied easily atm

**Review by rderfler — COMMENTED** [review]
Just a few thoughts about augment descriptions

File: src/Augmentation/data/AugmentationCreator.tsx [hunk 2]
```diff
@@ -95,6 +96,162 @@ function getRandomBonus(): any {
...
+    factions: [FactionNames.Infiltrators],
+  }),
+  new Augmentation({
+    name: AugmentationNames.CorporationManagementImplant,
+    repCost: 25e1,
+    moneyCost: 1e6,
+    infiltration_rep_mult: 2.5,
+    info:
+      "As time went on corporations realized that managers were redundant if they could be replaced by AI chips " +
+      "implanted directly in the brain, and so the this implant was developed which could analyse the users brain " +
```
[the remark this answers is not in the document]
Comment (phyzical) [reply src/Augmentation/data/AugmentationCreator.tsx:117]: will update 👍

File: src/Augmentation/data/AugmentationCreator.tsx [hunk 3]
```diff
@@ -95,6 +96,162 @@ function getRandomBonus(): any {
...
+      "of time begins to slow down allowing you to take more time to do things than the average person",
+    factions: [FactionNames.Infiltrators],
+  }),
+  new Augmentation({
+    name: AugmentationNames.BitaniumArmorAlloy,
+    repCost: 5e2,
+    moneyCost: 1e6,
+    infiltration_damage_reduction_mult: 0.7,
+    info:
+      `Deep in the mines of ${CityName.Ishima} miners found a strange new material, now known as bitanium after ` +
```
[the remark this answers is not in the document]
Comment (phyzical) [reply src/Augmentation/data/AugmentationCreator.tsx:158]: will update 👍

File: src/Augmentation/data/AugmentationCreator.tsx [hunk 4]
```diff
@@ -95,6 +96,162 @@ function getRandomBonus(): any {
...
+      "many iterations of experimenting it was found to be exceptional at increasing ones amour at absorbing " +
```
[the remark this answers is not in the document]
Comment (phyzical) [reply src/Augmentation/data/AugmentationCreator.tsx:159]: will update 👍

File: src/Augmentation/data/AugmentationCreator.tsx [hunk 5]
```diff
@@ -95,6 +96,162 @@ function getRandomBonus(): any {
...
+    factions: [FactionNames.Infiltrators],
+  }),
+  new Augmentation({
+    name: AugmentationNames.DyslexiaModule,
+    repCost: 1e2,
+    moneyCost: 1e6,
+    info:
+      "A module initially developed to reverse the disability of dyslexia, But during human trials it was found " +
+      "to actually cause dyslexia, despite the fact scientists decided to sell it anyway. Who would want " +
+      "to install something like that... i guess it has the added benefit of qualifying for a disability card",
```
[the remark this answers is not in the document]
Comment (phyzical) [reply src/Augmentation/data/AugmentationCreator.tsx:213]: ah good call 👍

**Pull request merged**